api_key_stamper/src/lib.rs

```diff
@@ -147,9 +168,116 @@ impl TurnkeyP256ApiKey {
     }
 }
 
+impl Stamp for TurnkeyP256ApiKey {
+    fn stamp(&self, body: &[u8]) -> Result<StampHeader, StamperError> {
+        let sig: P256Signature = self.signing_key.sign(body);
+        let stamp = TurnkeyApiStamp {
+            public_key: hex::encode(self.compressed_public_key()),
+            signature: hex::encode(sig.to_der()),
+            scheme: SIGNATURE_SCHEME_P256.to_string(),
+        };
+        let json_stamp = serde_json::to_string(&stamp).unwrap();
+        Ok(StampHeader {
+            name: API_KEY_STAMP_HEADER_NAME,
+            value: BASE64_URL_SAFE_NO_PAD.encode(json_stamp.as_bytes()),
+        })
+    }
+}
+
+/// Represents a Turnkey API key using the **secp256k1** curve.
+#[derive(Debug, PartialEq)]
+pub struct TurnkeySecp256k1ApiKey {
+    signing_key: K256SigningKey,
+}
+
+impl TurnkeySecp256k1ApiKey {
+    /// Generates a new Secp256k1 API key.
+    pub fn generate() -> Self {
+        let signing_key = K256SigningKey::random(&mut OsRng);
+        Self { signing_key }
+    }
+
+    /// Creates a new API key from private key bytes.
+    pub fn from_bytes<B: AsRef<[u8]>>(
+        private_key_bytes: B,
+        public_key_bytes: Option<B>,
+    ) -> Result<Self, StamperError> {
+        let signing_key = K256SigningKey::from_bytes(private_key_bytes.as_ref().into())
+            .map_err(|e| StamperError::InvalidPrivateKeyBytes(e.to_string()))?;
+
+        if let Some(pub_bytes) = public_key_bytes {
+            let expected = K256VerifyingKey::from_sec1_bytes(pub_bytes.as_ref())
+                .map_err(|e| StamperError::InvalidPublicKeyBytes(e.to_string()))?;
+            let actual = signing_key.verifying_key();
+            if expected != *actual {
+                return Err(StamperError::PublicKeyMismatch(
+                    hex::encode(expected.to_encoded_point(true).as_bytes()),
+                    hex::encode(actual.to_encoded_point(true).as_bytes()),
+                ));
+            }
+        }
+
+        Ok(Self { signing_key })
+    }
+
+    /// Load an API key from hex-encoded strings (with optional public key check).
+    pub fn from_strings<S: AsRef<str>>(
+        private_key: S,
+        public_key: Option<S>,
+    ) -> Result<Self, StamperError> {
+        let private_key_bytes = hex::decode(private_key.as_ref())
+            .map_err(|e| StamperError::HexDecode(e.to_string()))?;
+
+        let public_key_bytes = if let Some(pk) = public_key {
+            Some(hex::decode(pk.as_ref()).map_err(|e| StamperError::HexDecode(e.to_string()))?)
+        } else {
+            None
+        };
+
+        Self::from_bytes(private_key_bytes, public_key_bytes)
+    }
+
+    /// Returns compressed SEC1 public key bytes
+    pub fn compressed_public_key(&self) -> Vec<u8> {
+        self.signing_key
+            .verifying_key()
+            .to_encoded_point(true)
+            .as_bytes()
+            .to_vec()
+    }
```

Comment:
This is all copy/pasted from the P256 key implementation; I think it would be good to either consolidate with traits, or copy (or augment) tests so we have coverage for both p256 and k256. Your choice :D